matchbox_socket/src/webrtc_socket/socket.rs

```diff
@@ -179,6 +183,19 @@ impl WebRtcSocketBuilder {
         self.config.keep_alive_interval = interval;
         self
     }
+
+    /// Sets authentication on request to the signaling server.
+    pub fn basic_authentication(
+        mut self,
+        username: impl Into<String>,
+        password: impl Into<String>,
+    ) -> Self {
+        let username = username.into();
+        let password = password.into();
+        let auth = STANDARD.encode(format!("{username}:{password}"));
+        self.config.authentication = Some(format!("Basic {auth}"));
```

Comment:
I'm wondering if we should just have API that can set arbitrary headers on the signalling connect request, and some way in the signaling server apji to access those.

Then `basic_authentication` and eventually `bearer_authentication` would just be convenience on top of that?

Reply:
Read the discussion in the other thread now... So we can't have headers on websocket connect :/

I guess the alternative here would be to actually parse the room url, and then provide builder methods to add key-values as query params.

Another advantage of that is that we could potentially protect people from accidentally passing tokens over unencrypted http by checking the scheme in the builder method.

Reply:
Another option is to change the protocol so the client makes the first move, and add a

```rust
Connect(Vec<String, String)>)
```

to `PeerRequest`... and perhaps pass that on to the signaling server callback where you can reject clients?

Reply:
I'm leaning more and more towards that at least to start with, we should just make mechanisms that *enable* authentication, not the actual authentication itself.

e.g. just make it convenient to pass key-values when connecting. and convenient to access and deny based on them in the server callback. Something like this:

```rust
let socket = WebRtcSocket::builder("url")
    .add_socket(ChannelConfig::reliable())
    .add_param("token", token)
    .add_param("host", true)
    .build();
```

```rust
// server example
on_connect(|peer_id, params| {
        let Some(Ok(token)) = params.get("token").map(decode_token) else {
            return false;
        };
        let wants_host = params.get<bool>("host").unwrap_or(false);

        // you have to complete registration to host a session, but you could still join one
        token.claims.registration_completed || !wants_host
}); 

fn decode_token<T: AsRef<String>>(token: T) -> Token<Claims> {
    // use argon/biscuit whatever scheme tools you need to check the token
}
```

then we could just have username/password auth and token auth as examples.

Reply:
This seems like a good approach to me, though still hesitant to even show username/password auth

Reply:
Yea I'm cool with that after 2nd thought.